cogs/stockmarket/market.py

```diff
@@ -0,0 +1,568 @@
+                        split_due_per_share = actions.iloc[-1][1]
+
+                        # order doesn't matter since companies always try to avoid paying out dividends/splits at the same time
+                        # pay out the dividend
+                        shares_owned = holding['amount']
+                        member_object = None  # expected to change
+
+                        # pay dividend
+                        if dividend_due_per_share > 0:
+                            guild_object = self.bot.get_guild(singular_guild_id)
+                            member_object = await guild_object.query_members(user_ids=singular_member_id)
+                            currency_name = await bank.get_currency_name(guild_object)
+                            dividend_due = ceil(dividend_due_per_share * shares_owned * 100)
+
+                            await bank.deposit_credits(member_object[0], dividend_due)
+                            try:
+                                await member_object[0].send(f"you have received a dividend of {dividend_due} {currency_name} from {ticker}!")
+                            except (discord.Forbidden, discord.NotFound):
+                                print(f"Failed to notify user about receiving a dividend {member_object[0].name} in guild {guild_object.name}")
+
+                        # pay split
+                        if split_due_per_share > 0:
+                            holding['amount'] += holding['amount'] * split_due_per_share
+                            try:
+                                await member_object[0].send(f"you have received a {split_due_per_share} split from {ticker}!")
+                            except (discord.Forbidden, discord.NotFound):
+                                print(f"Failed to notify user about receiving a split {member_object[0].name} in guild {guild_object.name}")
+
+                        # for history
+                        member_dict['transactions'].append({
+                            'type': 'event',
+                            ticker: [most_recent_unix],
+                            'date': datetime.utcfromtimestamp((int)(time.time())).strftime('%Y-%m-%d %H:%M:%S'),
+                            'dividend_amount': ceil(dividend_due_per_share * shares_owned * 100),
+                            'split_ratio': split_due_per_share
+                        })
+
+    @commands.group()
+    async def market(self, ctx):
+        """Manage your Money! Manage your Life! Invest into the FUTURE!\n
+        This collection of commands allow you to invest your hard earned money,
+        into any publicly traded company around the world! #GetRichSlow
+        """
+
+    @market.command(name="sell")
+    async def sell_stock(self, ctx, ticker: str, amount_to_sell: int=0,  user_id: discord.Member=None):
+        """Sell a user-specified amount of a stock"""
+        ticker = ticker.upper()
+
+        if user_id is None:
+            user_id = ctx.author.id

```

Comment:
You declared discord.Member for user_id, please use `ctx.author` or change the type hint.